### PR TITLE
Fix exit non-zero exit codes when running as pid1

### PR DESCRIPTION
**Pull request by thockin**
(v4 branch)

Prior to this we would swallow the exit code and always exit(0).

### CODE REVIEW

File: pkg/pid1/pid1.go [hunk 1]
```diff
@@ -11,60 +11,73 @@ import (
...
 		}
 	}
+	return 0, fmt.Errorf("signal handler terminated unexpectedly")
 }
 
-// sigchld handles a SIGCHLD.  This will return true when firstborn exits.  In
-// case of truly unknown errors it will panic.
-func sigchld(firstborn int) bool {
+// sigchld handles a SIGCHLD.  This will return true only when firstborn exits.
+// For any other process thihs will return false and a 0 status.
```
Comment (stp-ip) [inline pkg/pid1/pid1.go:63]:
```suggestion
// For any other process this will return false and a 0 status.
```

**Review by stp-ip — APPROVED** [review]
/approve

Small nit/typo here as well

**Comment by thockin** [comment]
Typo fixed

**Pull request merged**